### PR TITLE
Add marker interface for event subscriber autoconfiguration

### PR DESCRIPTION
**Pull request by alterphp**
Totally copied from https://github.com/doctrine/DoctrineBundle/pull/1119 !

This is first step in order to split Doctriner EventSubscriber autoconfiguration and then resolve conflicts introduced in https://github.com/KnpLabs/DoctrineBehaviors/blob/master/src/Bundle/DependencyInjection/DoctrineBehaviorsExtension.php#L29

### CODE REVIEW

**Comment by alterphp** [comment]
Not sure if line should be adapted too https://github.com/doctrine/DoctrineMongoDBBundle/pull/623/files#diff-21bb4bdcf714a43b98948d9c96f74a1fR96

**Comment by malarzm** [comment]
@alterphp thanks for the PR and sorry for the long wait!

> Not sure if line should be adapted too https://github.com/doctrine/DoctrineMongoDBBundle/pull/623/files#diff-21bb4bdcf714a43b98948d9c96f74a1fR96

What line are you referring to?

Also would you mind adding a note to https://github.com/doctrine/DoctrineMongoDBBundle/blob/master/Resources/doc/events.rst? You may take a page from https://symfony.com/doc/master/bundles/DoctrineMongoDBBundle/first_steps.html#service-repositories

**Comment by malarzm** [comment]
@alterphp I'm afraid a conflict popped up, can you please rebase your changes atop current `master`? Also once you'll be rebasing, please squash your commits into 1.

**Comment by alterphp** [comment]
Yes, i'm not done, i had to switch to m'y dad duties

Le dim. 22 mars 2020 à 13:52, Maciej Malarz <notifications@github.com> a
écrit :

**Comment by alterphp** [comment]
> > Not sure if line should be adapted too https://github.com/doctrine/DoctrineMongoDBBundle/pull/623/files#diff-21bb4bdcf714a43b98948d9c96f74a1fR96
> What line are you referring to?

@malarzm I do not remember what I was thinking... Sorry. I htink it's ok now !

File: Resources/doc/events.rst [hunk 1]
```diff
@@ -63,7 +63,11 @@ when its event is dispatched.
 Event Subscribers
 -----------------
 
-Use the ``doctrine_mongodb.odm.event_subscriber`` tag
+Implement ``Doctrine\Bundle\MongoDBBundle\EventSubscriber\EventSubscriberInterface``
```
Comment (greg0ire) [inline Resources/doc/events.rst]:
```suggestion
Implementing ``Doctrine\Bundle\MongoDBBundle\EventSubscriber\EventSubscriberInterface``
```

Comment (alterphp) [reply Resources/doc/events.rst]: Why replacing Use by Implement**ing** ?

Comment (greg0ire) [reply Resources/doc/events.rst]: You can replace it with implement, but to be grammatically correct you will have to continue with "and autoconfiguration to automatically register"

Comment (alterphp) [reply Resources/doc/events.rst]: Ok, my explanation was incomplete. I was not telling to use autoconfiguration but that autoconfiguration will do the job. But you're right, I should explicitely tell to use autoconfigure.

Comment (greg0ire) [reply Resources/doc/events.rst]: I was more trying on "to" vs "will" here, not sure if that came through

Comment (alterphp) [reply Resources/doc/events.rst]: Is it better now ?

Comment (greg0ire) [reply Resources/doc/events.rst]: It sure is 👍

**Comment by malarzm** [comment]
Thanks a lot @alterphp!

> Yes, i'm not done, i had to switch to m'y dad duties

Sorry then, got a notification about a commit and acted on it :)

**Comment by alterphp** [comment]
No worries, you could not guess !

**Pull request merged**